cli/azd/pkg/project/framework_service_docker.go

```diff
@@ -158,6 +159,8 @@ func (p *dockerProject) Requirements() FrameworkRequirements {
 
 // Gets the required external tools for the project
 func (p *dockerProject) RequiredExternalTools(context.Context) []tools.ExternalTool {
+	// TODO(ellismg): When using remoteBuild, we shouldn't need docker. We need the service config here.
```

Comment:
Yes - i had this same thought.  For optional dependencies based on `azure.yaml` configuration we need to be able to inspect the `ServiceConfig` to know what is required.

Reply:
I will explore a change that moves this from `RequiredExternalTools(context.Context)` to `RequiredExternalTools(context.Context, project.ServiceConfig)` to get us in a place where we can do this.